wire/test/msgstest.go

```diff
@@ -46,8 +46,12 @@ func AuthMsgsTest(t *testing.T, serializerTest func(t *testing.T, msg wire.Msg))
 
 // newRandomstring returns a random ascii string of length between minLen and
 // minLen+maxLenDiff.
-func newRandomString(rng *rand.Rand, minLen, maxLenDiff int) string {
+func newRandomASCIIString(rng *rand.Rand, minLen, maxLenDiff int) string {
```

Comment:
You forgot to rename in the function comment.

Could it be a problem if protobuf only supports ASCII?

Reply:
I will rename in the comment.

The reason behind this change is that, protobuf encodes strings using UTF-8 encoding and hence supports all unicode characters. However, not all byte arrays are valid unicode strings (see [here](https://en.wikipedia.org/wiki/UTF-8#Encoding)) and protobuf errors when an invalid unicode string is passed to it.

Generating a random unicode string is a bit tricky, so I have used random ASCII string for tests.

Reply:
With perunio encoding, this was not a problem because `io.WriteString`, by default, treats the string as byte array.

Reply:
What i meant is that now the behavior of some messages is different for different encodings. Probably not a problem in most cases, but in an edge case somebody might try to encode some information into this message as bytes which will work with the perunio encoding, but not with the protobuf encoding.

Reply:
Okay, now I get your point. I can see two possible solutions:

1. If these fields can hold arbitrary binary data, then we can change its type in protobuf adapter from string to byte array. Then, this restriction does not apply.
2. If these fields can hold only string data, then perunio adapter could probably validate the string, before encoding.

What do you think @matthias ?

Reply:
OK, maybe we can add a comment to the type that this is expected to be a unicode string.

Reply:
Added the comment. See these two commits: 57e48716 and f08fb808.